src/main/java/codechicken/nei/recipe/GuiRecipe.java

```diff
@@ -388,9 +388,16 @@ public void mouseScrolled(int i) {
         // the value of the height field will be different from in other mouseover methods, which
         // could be confusing...
 
+        if (recipeTabs.mouseScrolled(i)) return;
```

Comment:
Hmm, might be good to put a comment here explaining that we're first checking if the mouse is in the tab area for scrolling, then we're checking the recipe handler for handling the scroll event, and finally we check if shift is down and scroll the tabs if so. I was thinking this mainly because it may not be obvious that lines 396-397 are actually doing the same thing as line 391, just with a shift-check instead of an area-check.